internal/controller/stas/workload_controller.go

```diff
@@ -118,6 +118,34 @@ func (r *PodReconciler) reconcilePod() reconcile.Func {
 	}
 }
 
+func (r *PodReconciler) imagePodIndex(ctx context.Context, pod *corev1.Pod) (map[string][]corev1.Pod, error) {
+	siblings := corev1.PodList{Items: []corev1.Pod{*pod}}
+	if c := metav1.GetControllerOfNoCopy(pod); c != nil {
+		if err := r.List(ctx, &siblings,
+			client.InNamespace(pod.Namespace),
+			client.MatchingFields{indexControllerUID: string(c.UID)},
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	index := map[string][]corev1.Pod{}
```

Comment:
Would it be possible to use a composite (logical) key in this map, with `containerName` and `image`.